external-crates/move/move-compiler/src/cfgir/optimize/known_jumps.rs

```diff
@@ -0,0 +1,145 @@
+// Copyright (c) The Move Contributors
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::{
+    cfgir::{
+        ast::remap_labels,
+        cfg::{MutForwardCFG, CFG},
+    },
+    hlir::ast::{BasicBlocks, Command, Command_, FunctionSignature, Label, SingleType, Var},
+    shared::unique_map::UniqueMap,
+};
+
+use std::collections::{BTreeMap, BTreeSet};
+
+/// returns true if anything changed
+pub fn optimize(
+    _signature: &FunctionSignature,
+    _locals: &UniqueMap<Var, SingleType>,
+    cfg: &mut MutForwardCFG,
+) -> bool {
+    let changed = optimize_(cfg.blocks_mut());
+    if changed {
+        cfg.recompute();
+        remap_in_order(cfg.start_block(), cfg.blocks_mut());
+    }
+    changed
+}
+
+fn optimize_(blocks: &mut BasicBlocks) -> bool {
+    let known_jumps = find_known_jumps(blocks);
+    optimize_known_jumps(blocks, known_jumps)
+}
+
+fn find_known_jumps(blocks: &BasicBlocks) -> BTreeMap<Label, Label> {
+    use Command_ as C;
+    let mut forwarded_jumps = BTreeMap::new();
+    for (label, block) in blocks {
```

Comment:
For fancy pants points
```suggestion
    let mut forwarded_jumps = blocks.iter().filter_map((block.len() == 1).then(|| 
     if let C::Jump { target, .. } = &block.last()?.value { Some((*label, *target)) }
     else { None }
     )).collect()
```
Some variation of this might be a bit simpler

Reply:
I'm ever so slightly happier with my rewrite. :)